aspnet/tutorials/first-web-api.rst

```diff
@@ -157,38 +157,63 @@ Here is an example HTTP response for the ``GetAll`` method::
+- If the ``[HttpGet]`` attribute has a template string, append that to the path. This sample doesn't use a template string.
+
+In the ``GetById`` method:
 
-For the ``GetById`` method,  "{id}" is a placeholder variable. In the actual HTTP request, the client will use the ID of the ``todo`` item. At runtime, when MVC invokes ``GetById``, it assigns the value of "{id}" in the URL the method's ``id`` parameter.
+.. code-block:: c#
+
+  [HttpGet("{id}", Name = "GetTodo")]
+  public IActionResult GetById(string id)
+
+.. review: Total rewrite, compare with original
+
+``"{id}"`` is a placeholder variable for the ID of the ``todo`` item. When ``GetById`` is invoked, it assigns the value of "{id}" in the URL to the method's ``id`` parameter.
+
+``Name = "GetTodo"`` creates an named route and allows you to link to this route in an HTTP Response. I'll explain it with an example later.
 
 Return values
 ^^^^^^^^^^^^^
 
-The ``GetAll`` method returns a CLR object. MVC automatically serializes the object to `JSON <http://www.json.org/>`__ and writes the JSON into the body of the response message. The response code for this method is 200, assuming there are no unhandled exceptions. (Unhandled exceptions are translated into 5xx errors.)
+The ``GetAll`` method returns an ``IEnumerable``. MVC automatically serializes the object to `JSON <http://www.json.org/>`__ and writes the JSON into the body of the response message. The response code for this method is 200, assuming there are no unhandled exceptions. (Unhandled exceptions are translated into 5xx errors.)
 
-In contrast, the ``GetById`` method returns the more general ``IActionResult`` type, which represents a generic result type. That’s because ``GetById`` has two different return types:
+In contrast, the ``GetById`` method returns the more general ``IActionResult`` type, which represents a wide range of return types. ``GetById`` has two different return types:
 
 - If no item matches the requested ID, the method returns a 404 error.  This is done by returning ``NotFound``.
-- Otherwise, the method returns 200 with a JSON response body. This is done by returning an `ObjectResult <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Mvc/ObjectResult/index.html>`_.
+- Otherwise, the method returns 200 with a JSON response body. This is done by returning an :dn:cls:`~Microsoft.AspNetCore.Mvc.ObjectResult`
+
+Launch the app
+^^^^^^^^^^^^^^^
+
+In Visual Studio, press ^F5 to launch the app. Visual Studio launches a browser and navigates to ``http://localhost:port/api/values``, where *port* is a randomly chosen port number. If you're using Chrome, Edge or Firefox, the *todo* data will be displayed. If you're using IE, IE will prompt to you open or save the *todo.json* file.
```

Comment:
This doesn't actually happen - bebcause it's going to the values controller -- /api/values URL, IE prompts for values.json, not todo.json